deploy/helm/templates/rbac.yaml

```diff
@@ -0,0 +1,82 @@
+{{- if .Values.serviceAccount.create -}}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "starboard-operator.serviceAccountName" . }}
+  labels:
+    {{- include "starboard-operator.labels" . | nindent 4 }}
+  {{- with .Values.serviceAccount.annotations }}
+  annotations:
+    {{- . | toYaml | nindent 4 }}
+  {{- end }}
+{{- end }}
+
+{{- if .Values.rbac.create }}
+---
+{{- /*
+Create (Cluster)Role and (Cluster)RoleBinding depending on if the Helm chart is
+installed in a namespace different from the targetNamespace.
+*/}}
+{{- $clusterWide := not (eq .Release.Namespace (tpl .Values.targetNamespaces .)) }}
+{{- $conditionalClusterPrefix := $clusterWide | ternary "Cluster" "" }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: {{ $conditionalClusterPrefix }}Role
```

Comment:
@danielpacak you wrote the following in #187:

> The Helm chart would also take care of creating RBAC objects based on operatorNamespace and operatorTargetNamespaces values.

1. I've made the `operatorNamespace` be in whatever namespace the Helm chart is installed, so no such configuration is exposed.
2. I've made `targetNamespaces` be a configuration of the Helm chart, which is by default the same as the namespace the Helm chart is installed.
3. The logic to decide of (Cluster)Role (Cluster)RoleBinding should be installed can be inspected above.

Does this seem like an okay implementation?